File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numToIndex = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int reverse = target - nums[i];
            if (numToIndex.containsKey(reverse)) {
                return new int[]{numToIndex.get(reverse), i};
            }
            numToIndex.put(nums[i], i);
        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-0.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        complete = [False, False, False]
        for x,y,z in triplets: 
            if x <= target[0] and y <= target[1] and z <= target[2]:
                if x == target[0]:
                    complete[0] = True
                if y == target[1]:
                    complete[1] = True
                if z == target[2]:
                    complete[2] = True
        return complete[0] and complete[1] and complete[2]
"""
(2,5,6), (1,4,4), (5,7,5) -> (5,4,6)

5 -> 2
4 -> 1


"""